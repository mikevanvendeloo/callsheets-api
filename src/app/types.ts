export type ActiveCallSheetTimer = {
  matchStartTime: string;
  activeItem: CallSheetItem;
  nextItem?: CallSheetItem;
  itemTimer: ItemTimer;
}

export type ItemTimer = {
  timerValue: string;
  timerColor: string;
  startTimestamp: string;
  endTimestamp: string;
}

export type CallSheetItem = {
  id: number;
  timeStart: string;
  durationInMinutes: number;
  timeEnd: string;
  title: string;
}

type CallSheet = {
    matchInfo: MatchInfo;
    schedule: Schedule;
}
export default CallSheet

export type MatchInfo = {
    league: string;
    title: string;
    date: string;
    startTime: string;
}
  
export type Schedule = {
    items: ScheduleItem[];
}

export type ScheduleItem = {
    id: number;
    timeStart: string;
    durationInMinutes: number;
    timeEnd: string;
    title: string;
}
  
export type ActiveCallSheet = {
    activeCallSheet: string;
    itemStartTimestamp: string
    activationTimestamp: string;
    activeItem: number;
    callSheet: CallSheet;
}

// export type CallSheetResponse = {
//     matchInfo: MatchInfoResponse;
//     schedule: ScheduleResponse;
// }

// export type ActiveCallSheetResponse = {
//     activeCallSheet: string;
//     itemStartTimestamp: string
//     activationTimestamp: string;
//     activeItem: number;
//     callSheet: CallSheetResponse;
// }


// export type MatchInfoResponse = {
//     league: string;
//     title: string;
//     date: string;
//     startTime: string;
// }
  
// export type ScheduleResponse = {
//     items: ScheduleItemResponse[];
// }

// export type ScheduleItemResponse = {
//     id: number;
//     timeStart: string;
//     durationInMinutes: number;
//     timeEnd: string;
//     title: string;
// }