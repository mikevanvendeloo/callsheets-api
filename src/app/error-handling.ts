import { type Request, type Response, type NextFunction } from 'express'
import type AppError from './api-error'
import { HttpStatusCode } from './api-error'

export const errorLogger = (
  error: Error,
  request: Request,
  response: Response,
  next: NextFunction,
): void => {
  console.log(`error ${error.message}`)
  next(error) // calling next middleware
}

// Error handling Middleware function reads the error message
// and sends back a response in JSON format
export const errorResponder = (
  error: AppError,
  request: Request,
  response: Response,
  next: NextFunction,
): void => {
  response.header('Content-Type', 'application/json')

  const status = error.statusCode ?? HttpStatusCode.BAD_REQUEST
  response.status(status).send(error.message)
}

// Fallback Middleware function for returning
// 404 error for undefined paths
export const invalidPathHandler = (
  request: Request,
  response: Response,
  next: NextFunction,
): void => {
  response.status(404)
  response.send('invalid path')
}
