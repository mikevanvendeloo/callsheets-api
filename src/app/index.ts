import express, { Application } from 'express'
import cors, { CorsOptions } from 'cors'
import Routes from './routes'
import requestLogging from './logging/requestLogging'
import path from 'path'

export const dataDirectory = path.join(__dirname, '../../data')
export const callSheetDirectory = path.join(dataDirectory, 'callsheets')

export default class Server {
  constructor(app: Application) {
    this.config(app)
    new Routes(app)
  }

  private config(app: Application): void {
    const corsOptions: CorsOptions = {
      origin: '*',
      methods: ['GET', 'POST', 'DELETE'],
      exposedHeaders: [
        'Content-Length',
        'origin',
        'X-Requested-With',
        'Content-Type',
        'Accept',
        'Authorization',
      ],
    }

    app.use(cors(corsOptions))
    app.use(express.json())
    app.use(express.urlencoded({ extended: true }))
    app.use(requestLogging)
  }
}
