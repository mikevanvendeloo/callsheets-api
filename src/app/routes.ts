import { Application } from 'express'
import callSheetRoutes from './callsheets/CallSheetRoutes'
import healthRoutes from './health/HealthRoutes'
import liveRoutes from './live/LiveRoutes'

export default class Routes {
  constructor(app: Application) {
    app.use('/health', healthRoutes)
    app.use('/api/live', liveRoutes)
    app.use('/api/callsheets', callSheetRoutes)
  }
}
