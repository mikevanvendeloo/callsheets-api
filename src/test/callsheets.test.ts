import request from 'supertest'
import app from '../server'

describe('Verify callsheet endpoint', () => {
  test('should return the list of callsheets that were uploaded', async () => {
    const res = await request(app).get('/api/callsheets')
    expect(res.body).toMatchSnapshot()
  })
})
