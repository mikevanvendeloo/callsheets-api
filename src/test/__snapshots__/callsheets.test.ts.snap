// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify callsheet endpoint should return the list of callsheets that were uploaded 1`] = `
[
  {
    "fileName": "17-11-2023__KL__Fortuna-Delta Logistiek - PKC-Vertom.json",
    "league": "KL",
    "matchDate": "17-11-2023",
    "matchTitle": "Fortuna-Delta Logistiek - PKC-Vertom",
  },
  {
    "fileName": "17-11-2023__resKL__Fortuna-Delta Logistiek - Blauw-Wit.json",
    "league": "resKL",
    "matchDate": "17-11-2023",
    "matchTitle": "Fortuna-Delta Logistiek - Blauw-Wit",
  },
]
`;
